pandas/core/internals/array_manager.py

```diff
@@ -659,24 +659,50 @@ def idelete(self, indexer):
 
     def iset(self, loc: Union[int, slice, np.ndarray], value):
         """
-        Set new item in-place. Does not consolidate. Adds new Block if not
-        contained in the current set of items
+        Set new column in-place (replaces (an) existing column(s)).
+
+        Parameters
+        ----------
+        loc : integer, slice or boolean mask
+            Positional location (already bounds checked)
+        value : array-like
         """
+        # single column -> single integer index
         if lib.is_integer(loc):
             # TODO normalize array -> this should in theory not be needed?
```

Comment:
what does "normalize" mean here?

Reply:
I mean the "extract_array" call could/should in theory not be needed (will replace normalize with extract)